service/kernel/data_management/finiteness_checker.cpp

```diff
@@ -129,6 +140,7 @@ DataType computeSumAVX512Impl(size_t nDataPtrs, size_t nElementsPerPtr, const Da
 
     daal::services::internal::TArray<DataType, avx512> partialSumsArr(nTotalBlocks);
     DataType * pSums = partialSumsArr.get();
+    if (!pSums) return getInf<DataType>();
```

Comment:
It`s not clear. Why we return inf in case of memory allocation error.

Reply:
If infinity is returned as sum, finiteness checker goes to next step (finiteness check of all elements). Comment added.

Reply:
seems status should be propagated here (by parameter for example )